ext/docker/runmount_buildpack.go

```
@@ -0,0 +1,125 @@
+package docker
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"time"
+
+	sous "github.com/opentable/sous/lib"
+	"github.com/opentable/sous/util/logging"
+	"github.com/opentable/sous/util/logging/messages"
+)
+
+type (
+	// RunmountBuildpack builds a container, runs it seperately to use docker mounts for cache and output,
+	// and builds final deploy container
+	RunmountBuildpack struct {
+		detected *sous.DetectResult
+	}
+)
+
+// NewRunmountBuildpack returns a new RunmountBuildpack
+func NewRunmountBuildpack() *RunmountBuildpack {
+	return &RunmountBuildpack{}
+}
+
+func readDockerfile() (string, error) {
+	b, err := ioutil.ReadFile("Dockerfile")
+	if err != nil {
+		return "", err
+	}
+	return string(b), nil
+}
+
+// Detect implements Buildpack on RunmountBuildpack
+func (rmbp *RunmountBuildpack) Detect(ctx *sous.BuildContext) (*sous.DetectResult, error) {
+	dfPath := filepath.Join(ctx.Source.OffsetDir, "Dockerfile")
+	if !ctx.Sh.Exists(dfPath) {
+		return nil, fmt.Errorf("%s does not exist", dfPath)
+	}
+
+	messages.ReportLogFieldsMessage("Runmount dockerfile detection", logging.DebugLevel, logging.Log, dfPath)
+
+	dockerContent, _ := readDockerfile()
```

We can't ignore this error, if the read fails the rest of this method is meaningless, so need to return early if err != nil.